File: Tercer_Cuatrimestre/dotnet/Aseguradora/Version_1/Aseguradora/Aplicacion/Entidades/Persona.cs
namespace Aplicacion;
public abstract class Persona
{
    public int ID { get; set; } = -1;
    public string Nombre { get; }
    public string Apellido { get; }
    public int DNI { get; }
    public long Telefono { get; set; }

    public Persona(string nombre, string apellido, int dni)
    {
        Nombre = nombre;
        Apellido = apellido;
        DNI = dni;
    }
    public Persona(string nombre, string apellido, int dni, long telefono) : this(nombre, apellido, dni)
    {
        Telefono = telefono;
    }
    public override string ToString()
    {
        string aux = ($" {Nombre} | {Apellido} | {DNI} | {Telefono}");
        return (aux);
    }
    public void setId(int id)
    {
        ID = id;
    }
}

File: Cuarto/Laboratorio/Prácticas/Práctica 4/Ej1f/src/Persona.java
public abstract class Persona implements Comparable<Persona> {
     String apellido;
     String nombre;
     int dni;

    public Persona(String apellido, String nombre, int dni) {
        this.apellido = apellido;
        this.nombre = nombre;
        this.dni = dni;
    }
    @Override
    public int compareTo(Persona o) {
        return apellido.compareTo(o.apellido);
    }

    @Override
    public String toString() {
        return "Persona{" +
                "apellido='" + apellido + '\'' +
                ", nombre='" + nombre + '\'' +
                ", dni=" + dni +
                '}';
    }
}
